packages/eas-cli/src/build/utils/repository.ts

```diff
@@ -83,6 +83,50 @@ export async function commitPromptAsync(
   });
 }
 
+export async function makeProjectMetadataFileAsync(
+  archivePath: string
+): Promise<{ path: string; size: number }> {
+  const spinner = ora('Creating project metadata file');
+  const timerLabel = 'makeProjectMetadataFileAsync';
+  const timer = setTimeout(
+    () => {
+      spinner.start();
+    },
+    Log.isDebug ? 1 : 1000
+  );
+  startTimer(timerLabel);
+
+  const metadataLocation = path.join(getTmpDirectory(), `${uuidv4()}-eas-build-metadata.json`);
+  const archiveContent: string[] = [];
+
+  try {
+    await tar.list({
+      file: archivePath,
+      onentry: (entry: tar.ReadEntry) => {
+        if (entry.type === 'File' && !entry.path.includes('.git')) {
```

Comment:
Should we switch to `startsWith('.git/')` or `startsWith('/.git/')` (I don't know which would be the correct option)? This way we'll also show `.gitignore`, `.github/`.